fixtures/pages-functions-with-routes-app/tests/index.test.ts

```diff
@@ -0,0 +1,87 @@
+import { spawn } from "child_process";
+import * as path from "path";
+import { fetch } from "undici";
+import type { ChildProcess } from "child_process";
+import type { Response, RequestInit } from "undici";
+
+const waitUntilReady = async (
+	url: string,
+	requestInit?: RequestInit
+): Promise<Response> => {
+	let response: Response | undefined = undefined;
+
+	while (response === undefined) {
+		await new Promise((resolvePromise) => setTimeout(resolvePromise, 500));
+
+		try {
+			response = await fetch(url, requestInit);
+		} catch {}
+	}
+
+	return response as Response;
+};
+
+const isWindows = process.platform === "win32";
+
+describe("Pages Functions with custom _routes.json", () => {
+	let wranglerProcess: ChildProcess;
+
+	beforeEach(() => {
+		wranglerProcess = spawn("npm", ["run", "dev"], {
+			shell: isWindows,
+			cwd: path.resolve(__dirname, "../"),
+			env: { BROWSER: "none", ...process.env },
+		});
+		wranglerProcess.stdout?.on("data", (chunk) => {
+			console.log(chunk.toString());
+		});
+		wranglerProcess.stderr?.on("data", (chunk) => {
+			console.log(chunk.toString());
+		});
+	});
+
+	afterEach(async () => {
+		await new Promise((resolve, reject) => {
+			wranglerProcess.once("exit", (code) => {
+				if (!code) {
+					resolve(code);
+				} else {
+					reject(code);
+				}
+			});
+			wranglerProcess.kill("SIGTERM");
+		});
+	});
+
+	it("renders static pages", async () => {
+		const response = await waitUntilReady("http://localhost:8776/");
+		const text = await response.text();
+		expect(text).toContain(
+			"Bonjour de notre projet ✨pages-functions-with-routes-app✨!"
+		);
+	});
+
+	it("runs each Pages Function when their corresponding URL path is hit", async () => {
+		let response = await waitUntilReady("http://localhost:8776/greeting/hello");
+		let text = await response.text();
+		expect(text).toMatch("Bonjour le monde!");
+
+		response = await waitUntilReady("http://localhost:8776/greeting/goodbye");
+		text = await response.text();
+		expect(text).toMatch("A plus tard alligator 👋");
+
+		/**
+		 * :sad_panda:
```

Comment:
Can we make a GitHub issue to track this feature parity in `wrangler pages dev`?

Reply:
+1! that was unexpected for me as well and meant to ask @jahands what we want to do abt it. I think this will create a lot of confusion for users if we leave it as is.

Reply:
Created an issue internally, good catch!